unittest/saidump/main.cpp

```diff
@@ -0,0 +1,55 @@
+#include <gtest/gtest.h>
+#include "saidump.cpp"
```

Comment:
please don't do that, don't include cpp files in another cpp files, i mentioned in comments that you would need to convert that saidump.cpp to library and then use taht library in the unittests, and saidump binary, then your test cases will be executed, since this file right now is treated as include not actual source, thats why is not covered even if tests are executed